src/spatial-index.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  DEFAULT_AGGREGATION_RES_LEVEL_H3,
+  DEFAULT_AGGREGATION_RES_LEVEL_QUADBIN,
+} from './constants-internal';
+import {ModelSource} from './models/model';
+import {AggregationOptions} from './sources/types';
+import {assert} from './utils';
+
+const DEFAULT_TILE_SIZE = 512;
+const QUADBIN_ZOOM_MAX_OFFSET = 4;
+
+export function getSpatialFiltersResolution({
+  source,
+  viewState,
+}: {
+  source: {
+    spatialDataType?: ModelSource['spatialDataType'];
+    dataResolution?: ModelSource['dataResolution'];
+    aggregationResLevel?: AggregationOptions['aggregationResLevel'];
+  };
```

Comment:
I think this part could be simplified as:

```suggestion
  source: Partial<ModelSource & AggregationOptions>;
```

Reply:
Done 👍